src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -13758,6 +13553,9 @@ private void LeaveQuery()
             _syntaxFactoryContext.QueryDepth--;
         }
 
+        private DisposableResetPoint GetDisposableResetPoint(bool resetOnDispose)
+            => new(this, resetOnDispose, GetResetPoint());
```

Comment:
The compiler codebase avoids implicitly-typed `new`, so that we can do textual search on `new Type(`.

Reply:
Find-References will find those cases.

Reply:
If something is disallowed, can you setup a rule to enforce that?

Reply:
Yes, a rule would be nice. I'm wondering what would be the most direct/simple way to set this up.